core/src/main/scala/org/apache/spark/network/netty/NettyBlockTransferService.scala

```diff
@@ -112,8 +117,18 @@ private[spark] class NettyBlockTransferService(
       val blockFetchStarter = new RetryingBlockFetcher.BlockFetchStarter {
         override def createAndStart(blockIds: Array[String], listener: BlockFetchingListener) {
           val client = clientFactory.createClient(host, port)
-          new OneForOneBlockFetcher(client, appId, execId, blockIds, listener,
-            transportConf, tempFileManager).start()
+          try {
+            new OneForOneBlockFetcher(client, appId, execId, blockIds, listener,
+              transportConf, tempFileManager).start()
+          } catch {
+            case e: IOException =>
+              if (driverEndPointRef.askSync[Boolean](IsExecutorAlive(execId))) {
```

Comment:
nit: add a safeguard when `driverEndPointRef` is null.

Reply:
> nit: add a safeguard when `driverEndPointRef` is null.
we use the default null value only in test. So, I think we may not need add safeguard when `driverEndPointRef` is null.

@cloud-fan

Reply:
then let's add an assert

Reply:
> then let's add an assert

done.

@cloud-fan